### PR TITLE
Created user model and controller

### PR DESCRIPTION
**Pull request by afarah613**

### CODE REVIEW

File: src/main/java/perkpack/controllers/UserController.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+public class UserController {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @PostMapping
+    public User createUser(@Valid @RequestBody User user)
+    {
+        return userRepository.save(user);
+
```
Comment (cyruscse) [inline src/main/java/perkpack/controllers/UserController.java]: Extra line here, delete this.

File: src/test/java/controllers/UserControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package controllers;
+
+public class UserControllerTest {
```
Comment (lavatahir) [inline src/test/java/controllers/UserControllerTest.java]: why is this empty

File: src/main/java/perkpack/models/User.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    @NotNull
+    @Email
+    @Size(min =1, max = 32)
+    private String email;
+
+    @Id
+    @GeneratedValue
+    private long id;
+
+    public User()
```
Comment (cyruscse) [inline src/main/java/perkpack/models/User.java]: We don't need this for H2 DB.

File: src/test/java/controllers/UserControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,4 @@
+package controllers;
```
Comment (cyruscse) [inline src/test/java/controllers/UserControllerTest.java]: Don't include this file, UserControllerTest already exists under perkpack/controllers/UserControllerTest

File: src/main/java/perkpack/models/User.java (hunk 3 above)
Comment (lavatahir) [reply src/main/java/perkpack/models/User.java]: Just found out that the tests rely on his empty constructor.

**Pull request merged**